src/proto/generation.proto

```diff
@@ -286,6 +286,7 @@ message Answer {
   uint64 created = 4;
   optional AnswerMeta meta = 6;
   repeated Artifact artifacts = 7;
+  optional double estimated_cost = 8;  // Only set if Request.dry_run is true.
```

Comment:
I'm partial to just calling this `cost` which would allow it to be used both for `dry_run` estimates as well as actual cost if desired.  `dry_run` already implies estimation / non-actual results.

Reply:
Units for this need to be documented somewhere as well ($ or credits).

Reply:
> I'm partial to just calling this cost which would allow it to be used both for dry_run estimates as well as actual cost if desired. dry_run already implies estimation / non-actual results.

I like that.

> Units for this need to be documented somewhere as well ($ or credits).

Definitely.  I'm waiting for feedback on which units we want to use here, so it's deliberately vague at the moment.

Reply:
I think we want credits here. And we should be able to return the `cost` of a `Request` in an `Answer` regardless of whether or not dry run is set.

Reply:
We definitely could.

Do y'all feel that in a multi-answer scenario, returning the cost on each Answer could be confusing?

Reply:
I think returning the cost with the respective answer would be the appropriate way of doing it and less confusing than returning the cost elsewhere.

Reply:
Where I see the confusion coming from is that we're computing the cost for a single request, but returning multiple answers.

Do the answers report their share of the cost, or do they report the total cost of the request?

Reply:
I suppose regardless of the exact desired behavior the cost needs to be within an answers object since that is the message used for the responses. Either each answer could have partial cost, or the total cost could be reported by the last answer sent back to the client.  I think either one could be appropriate. Regardless of that implementation detail I think we want to move forward with labeling this `cost` and per other discussions using `credits` as the unit of measure here.